### PR TITLE
Initial fixup of the VSCode extension

### LINKED ISSUE #309: The vscode extension should be cleaned up and released
I think it would be nice if the vscode extension was released and polished and usable for Buck2. I have it installed locally, but it's a chore especially when using features like remote vscode.

What would be needed for this? Ideally it would be released through GitHub as well, which would be really nice to help people contribute.

A tricky thing is how much buck2-specific "stuff" should be in it. The extension is distributed with the starlark package, but right now buck2 is the main user. Ideally, buck2 *specific* integration would include things like running or listing build targets...

### PR DESCRIPTION
**Pull request by mempler**
This PR is just fixing up and cleaning up the old VSCode extension with an at least initial attempt to fix #309 and #112 

This will make the extension a bit more useful than it was before.
allthough debugging capabilities do not work right now.

Much more work in the future will be required to properly hook up hovering and debugging.

### CODE REVIEW

File: starlark-rust/vscode/.vscode/extensions.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+  // See http://go.microsoft.com/fwlink/?LinkId=827846
+  // for the documentation about the extensions.json format
+  "recommendations": ["dbaeumer.vscode-eslint", "amodio.tsl-problem-matcher"]
```
Comment (ndmitchell) [inline starlark-rust/vscode/.vscode/extensions.json:4]: Is this a good thing to include? Do we really want to recommend it? Or is this normal for a vscode package?

File: starlark-rust/vscode/package.json [hunk 2]
```diff
@@ -49,59 +46,21 @@
...
         "grammars": [
             {
                 "language": "starlark",
                 "scopeName": "source.starlark",
-                "path": "./syntaxes/starlark.tmLanguage.json"
-            }
-        ],
-        "breakpoints": [
-            {
-                "language": "starlark"
```
Comment (ndmitchell) [inline starlark-rust/vscode/package.json:64]: Why delete the debugger/breakpoint features?

File: starlark-rust/vscode/package.json [hunk 3]
```diff
@@ -49,59 +46,21 @@
...
-                ]
+                "path": "./syntax/starlark.tmLanguage.json"
             }
         ],
         "configuration": {
             "properties": {
                 "starlark.lspPath": {
                     "type": "string",
-                    "default": "starlark",
+                    "1fault": "buck2",
```
Comment (ndmitchell) [inline starlark-rust/vscode/package.json:63]: Intentional? `1fault`

**Review by ndmitchell — COMMENTED** [review]
Thanks for the changes. Sorry we didn't notice them sooner - the notifications went to someone on holiday.

As for the changes, it seems there is a switch from spaces to tabs - I thought the expectation was spaces everywhere now? It also seems to use `buck2` rather than `starlark` as the driver. I'm honestly not sure what to do about that - we deliberately don't want to couple the starlark-rust project to Buck2, but it is the biggest user. Maybe there should be a switch? Maybe it should try and find if buck2 or starlark is on the path?

Is the nix thing entirely optional? By removing the .lock directories, doesn't that risk the breakage as packages upgrade?

File: starlark-rust/vscode/package.json (hunk 2 above)
Comment (mempler) [reply starlark-rust/vscode/package.json:64]: Due to lack of implementation within `buck2 lsp` I noticed later that they were under a different command. Didn't had time yet to reimplement them

**Comment by mempler** [comment]
> Thanks for the changes. Sorry we didn't notice them sooner - the notifications went to someone on holiday.

happens


> As for the changes, it seems there is a switch from spaces to tabs - I thought the expectation was spaces everywhere now? It also seems to use `buck2` rather than `starlark` as the driver. I'm honestly not sure what to do about that - we deliberately don't want to couple the starlark-rust project to Buck2, but it is the biggest user. Maybe there should be a switch?

There were some talks about this in the mentioned issue, thus I thought that was the right direction. I obviously do not have access to any internal communication that happen.

> Maybe it should try and find if buck2 or starlark is on the path?

Right now, it uses the one that is already in the `PATH` variable.

> Is the nix thing entirely optional? By removing the .lock directories, doesn't that risk the breakage as packages upgrade?

The nix thing is indeed entirely optional. I used nixos at the time of implementation.

File: starlark-rust/vscode/package.json (hunk 3 above)
Comment (mempler) [reply starlark-rust/vscode/package.json:63]: Typo

**Comment by mempler** [comment]
Due to lack of interest from my side, I'll close this PR. however I'll keep my fork and branch around in case anyone wants to Pickup on it.

**Comment by mempler** [comment]
I've decided to publish my own third-party extension, it's source available under https://github.com/PeerStudios/buck2-lsp-adapter and published under https://marketplace.visualstudio.com/items?itemName=PeerStudios.buck2-lsp-adapter which requires buck2 itself instead of starlark-rust. similarly to what i did in this PR.

**Pull request closed without merging**